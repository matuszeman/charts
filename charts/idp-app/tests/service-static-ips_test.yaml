suite: service-static-ips
templates:
  - service.yaml
values:
  - values/minimum.yaml
tests:
  - it: should generate EndpointSlices
    values:
      - values/service.yaml
      - values/deployment-static-ips.yaml
    asserts:
      - matchSnapshot: {}
  - it: when multi deployment
    values:
      - values/service.yaml
      - values/deployment-static-ips.yaml
      - values/multi.yaml
    asserts:
      - matchSnapshot: { }
  - it: when multi deployment with overwrite
    values:
      - values/service.yaml
      - values/deployment-static-ips.yaml
    set:
      deployment:
        staticIps:
          - "192.168.88.4"
          - "192.168.88.5"
        multi:
          one: {}
          two:
            staticIps:
              - 192.168.100.1
    asserts:
      - matchSnapshot: { }
