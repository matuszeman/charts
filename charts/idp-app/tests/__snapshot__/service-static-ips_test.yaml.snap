should generate EndpointSlices:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME
    spec:
      ports:
        - name: app
          port: 80
          protocol: TCP
          targetPort: 8123
      type: ClusterIP
  2: |
    addressType: IPv4
    apiVersion: discovery.k8s.io/v1
    endpoints:
      - addresses:
          - 192.168.88.4
      - addresses:
          - 192.168.88.5
    kind: EndpointSlice
    metadata:
      labels:
        endpointslice.kubernetes.io/managed-by: idp-app
        kubernetes.io/service-name: RELEASE-NAME
      name: RELEASE-NAME-static
    ports:
      - name: app
        port: 8123
        protocol: TCP
when multi deployment:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-one
    spec:
      ports:
        - name: app
          port: 80
          protocol: TCP
          targetPort: 8123
      type: ClusterIP
  2: |
    addressType: IPv4
    apiVersion: discovery.k8s.io/v1
    endpoints:
      - addresses:
          - 192.168.88.4
      - addresses:
          - 192.168.88.5
    kind: EndpointSlice
    metadata:
      labels:
        endpointslice.kubernetes.io/managed-by: idp-app
        kubernetes.io/service-name: RELEASE-NAME-one
      name: RELEASE-NAME-one-static
    ports:
      - name: app
        port: 8123
        protocol: TCP
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-two
    spec:
      ports:
        - name: app
          port: 80
          protocol: TCP
          targetPort: 8123
      type: ClusterIP
  4: |
    addressType: IPv4
    apiVersion: discovery.k8s.io/v1
    endpoints:
      - addresses:
          - 192.168.88.4
      - addresses:
          - 192.168.88.5
    kind: EndpointSlice
    metadata:
      labels:
        endpointslice.kubernetes.io/managed-by: idp-app
        kubernetes.io/service-name: RELEASE-NAME-two
      name: RELEASE-NAME-two-static
    ports:
      - name: app
        port: 8123
        protocol: TCP
when multi deployment with overwrite:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-one
    spec:
      ports:
        - name: app
          port: 80
          protocol: TCP
          targetPort: 8123
      type: ClusterIP
  2: |
    addressType: IPv4
    apiVersion: discovery.k8s.io/v1
    endpoints:
      - addresses:
          - 192.168.88.4
      - addresses:
          - 192.168.88.5
    kind: EndpointSlice
    metadata:
      labels:
        endpointslice.kubernetes.io/managed-by: idp-app
        kubernetes.io/service-name: RELEASE-NAME-one
      name: RELEASE-NAME-one-static
    ports:
      - name: app
        port: 8123
        protocol: TCP
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-two
    spec:
      ports:
        - name: app
          port: 80
          protocol: TCP
          targetPort: 8123
      type: ClusterIP
  4: |
    addressType: IPv4
    apiVersion: discovery.k8s.io/v1
    endpoints:
      - addresses:
          - 192.168.100.1
    kind: EndpointSlice
    metadata:
      labels:
        endpointslice.kubernetes.io/managed-by: idp-app
        kubernetes.io/service-name: RELEASE-NAME-two
      name: RELEASE-NAME-two-static
    ports:
      - name: app
        port: 8123
        protocol: TCP
